modules/social_features/social_activity/social_activity.module

```diff
@@ -234,3 +235,44 @@ function social_activity_views_post_build(ViewExecutable $view) {
     }
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function social_activity_preprocess_block(&$variables) {
```

Comment:
I think it would be better to stay consistent with our current code style about new lines and etc. And also do not duplicate of `$variables['content']['#name']`, `$node->bundle()`.
Here's my proposal:
```
function social_activity_preprocess_block(&$variables) {
  $route_match = \Drupal::routeMatch();
  $allowed_routes = [
    'entity.node.canonical',
    'layout_builder.overrides.node.view',
  ];
  $allowed_bundles = [
    'dashboard',
    'landing_page',
  ];
  $allowed_content_names = [
    'activity_stream',
    'community_activities',
    'activity_stream_profile',
    'activity_stream_group',
  ];

  if (($node = $route_match->getParameter('node')) && $node instanceof NodeInterface) {
    if (
      in_array($route_match->getRouteName(), $allowed_routes) &&
      in_array($node->bundle(), $allowed_bundles) &&
      $variables['base_plugin_id'] === 'views_block' &&
      isset($variables['content']['#name']) &&
      in_array($variables['content']['#name'], $allowed_content_names)
    ) {
      if ($node->bundle() === 'dashboard') {
        $variables['attributes']['class'][] = 'block-dashboard-activity-stream card';
      }
      /** @var \Drupal\views\ViewExecutable $view */
      $view = $variables['content']['#view'];
      $variables['content'] = [
        '#create_placeholder' => TRUE,
        '#lazy_builder' => [
          'social_activity.lazy_builder:viewsLazyBuild',
          [
            $variables['content']['#name'],
            $variables['content']['#display_id'],
            $node->bundle(),
            $view->getItemsPerPage(),
          ],
        ],
      ];
    }
  }
}
```

Also, I think it would be good to describe conditions a bit.

Reply:
Also, after the changes, the code will become more readable.